File: backend/src/main/java/com/cs353/backend/model/entities/JobOpening.java
package com.cs353.backend.model.entities;


import com.cs353.backend.Enum.EmploymentStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class JobOpening {
    private int jobOpeningID;
    private String employmentStatus;
    private String explanation;
    private LocalDateTime dueDate;
    private String rolePro;
    private String location;
    private String workType;
}

File: frontend/src/components/JobOpening.js
import React from 'react';

const JobOpening = ({ jobOpeningId, employmentStatus, explanation, dueDate, rolePro, location, workType }) => {
    return (
        <div style={{
            border: '1px solid #ccc',
            borderRadius: '4px',
            padding: '16px',
            marginBottom: '16px',
            backgroundColor: '#fff',
            boxShadow: '0 2px 4px rgba(0, 0, 0, 0.1)',
            textAlign: 'center'
        }}>
            <h3>{employmentStatus}</h3>
            <p style={{
                border: '1px solid #ccc',
                borderRadius: '4px',
                fontSize: "20px",
                backgroundColor: '#fff',
                marginLeft: "200px",
                marginRight: "200px",
                textAlign: "left"
            }}>Explanation: {explanation}</p>
            <p style={{
                marginLeft: "200px",
                marginRight: "200px",
                textAlign: "left"}}>Due Date: {dueDate}</p>
            <p style={{
                marginLeft: "200px",
                marginRight: "200px",
                textAlign: "left"}}>Role Pro: {rolePro}</p>
            <p style={{
                marginLeft: "200px",
                marginRight: "200px",
                textAlign: "left"}}>Location: {location}</p>
            <p style={{
                marginLeft: "200px",
                marginRight: "200px",
                textAlign: "left"}}>Work Type: {workType}</p>
        </div>
    );
};

export default JobOpening;
